### PR TITLE
Re-implement onnx old-style extractors with extractor extensions

### PR DESCRIPTION
**Pull request by evolosen**
Added new extractor classes for onnx such as ReshapeFrontExtractor, IdentityFrontExtractor, ConcatFrontExtractor, BatchNormalizationExtractor. Operation BN was modified and renamed as BatchNormInference.

Ticket: #44457

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests: N/A
* [x]  Framework operation tests: N/A
* [x]  Transformation tests: N/A
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py: N/A
* [x]  Model Optimizer IR Reader check: Done

Documentation:
* [x]  Supported frameworks operations list: N/A
* [x]  Supported **public** models list: N/A
* [x]  New operations specification: N/A
* [x]  Guide on how to convert the **public** model: N/A
* [x]  User guide update: N/A

### CODE REVIEW

File: model-optimizer/extensions/ops/BatchNormInference.py [hunk 1]
```diff
@@ -18,18 +18,23 @@
...
             'in_ports_count': 5,
             'out_ports_count': 1,
-            'infer': None
+            'infer': self.infer
         }, attrs)
+    @staticmethod
+    def infer(node):
+        node.out_port(0).data.set_shape(node.in_port(0).data.get_shape())
+        if node.in_port(0).data.get_value() is not None:
+            node.out_port(0).data.set_value(node.in_port(0).data.get_value())
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/BatchNormInference.py:40]: The value propagation is wrong. Please, remove it

File: model-optimizer/extensions/ops/BatchNormInference.py [hunk 2]
```diff
@@ -18,18 +18,23 @@
...
-    op = 'BN'
-    enabled = True
+    op = 'batchNormInference'
+    enabled = False
 
     def __init__(self, graph: Graph, attrs: dict):
         super().__init__(graph, {
-            'type': None,
-            'op': __class__.op,
+            'type': self.op,
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/BatchNormInference.py:30]: Please, leave `type` equal to `None` for now. We don't want to emit IR with this operation yet.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/extensions/ops/BatchNormInference.py (hunk 1 above)
Comment (evolosen) [reply model-optimizer/extensions/ops/BatchNormInference.py:40]: fixed

File: model-optimizer/extensions/ops/BatchNormInference.py (hunk 2 above)
Comment (evolosen) [reply model-optimizer/extensions/ops/BatchNormInference.py:30]: fixed

**Review by lazarevevgeny — APPROVED** [review]

**Review by pavel-esir — APPROVED** [review]
Looks good to me.

**Pull request merged**